src/sim/include/AbstractDriver.h

```diff
@@ -1198,6 +1225,58 @@ namespace MFM
       m_externalConfig.Write(fs);
       fs.Close();
     }
+    
+    void SaveGridDetail(OurGrid& grid, const char* filename, bool saveBaseState)
+    {
+
+      LOG.Message("Saving detail to: %s", filename);
+      FILE* fp = fopen(filename, "w");
+      FileByteSink fs(fp);
+      
+      const u32 gridWidth = grid.GetWidthSites();
+      const u32 gridHeight = grid.GetHeightSites();
+      const bool isStaggeredGrid = grid.IsGridLayoutStaggered();
+
+      for(s32 y = 0; y < (s32)gridHeight; y++)
+        {
+	  for(s32 x = 0; x < (s32)gridWidth; x++)
+  	  {
+	    SPoint siteInGrid;
+	    siteInGrid.SetX(x);
+	    siteInGrid.SetY(y);
+  	    if(isStaggeredGrid && !grid.IsGridCoord(siteInGrid)) continue;
+
+            const u32 printFlags =
+              UlamClassPrintFlags::PRINT_MEMBER_NAMES |
+              UlamClassPrintFlags::PRINT_MEMBER_VALUES |
+              UlamClassPrintFlags::PRINT_RECURSE_QUARKS;
+	    
+            OString512 buff;
```

Comment:
Not sure if this is where the data is getting truncated but if so could bump up to OString1024, 2048, 4096 or 8192..

Reply:
Makes sense if the UlamElement Print method accepts it. Do you figure this would still be required if there's a separate method to output JSON from the UlamElement<EC> directly?

Reply:
Most such methods take ByteSink&, so any of the OString's will go.  Plus other stuff like a FileByteSink as well, so yeah, might not need the buffer at all in that case.

Reply:
I changed this to OString2048. It seems to work as expected (now using the more-jsony output).
I think it's funny that trying to output valid JSON is more fragile than the prior solution. Before, the separate deserialization step meant I had to deal with the X-truncated string and would produce useful, incomplete output. Now, if the JSON gets truncated, any valid linter should reject the output and break anything depending on the formatting. 
It's fine. We'll never need more than this and nothing can possibly go wrong.